packages/openapi3/test/status-codes.test.ts

```diff
@@ -0,0 +1,60 @@
+import { expectDiagnostics } from "@typespec/compiler/testing";
+import { deepStrictEqual } from "assert";
+import { diagnoseOpenApiFor, openApiFor } from "./test-host.js";
+
+describe("openapi3: response status codes", () => {
+  async function expectStatusCodes(code: string, statusCodes: string[]) {
+    const res = await openApiFor(code);
+    deepStrictEqual(Object.keys(res.paths["/"].get.responses), statusCodes);
+  }
+  it("map single status code", async () => {
+    await expectStatusCodes(
+      `
+      op read(): {@statusCode _: 200, content: string};
+      `,
+      ["200"]
+    );
+  });
+
+  it("map multiple status code", async () => {
+    await expectStatusCodes(
+      `
+      op read(): {@statusCode _: 200 | 201 | 204, content: string};
+      `,
+      ["200", "201", "204"]
+    );
+  });
+
+  it("map simple status code range", async () => {
+    await expectStatusCodes(
+      `
+      op read(): {@minValue(400) @maxValue(499) @statusCode _: int32, content: string};
+      `,
+      ["4XX"]
+    );
+  });
+
+  it("map status code range going over multiple hundreds", async () => {
+    await expectStatusCodes(
+      `
+      op read(): {@minValue(400) @maxValue(599) @statusCode _: int32, content: string};
+      `,
+      ["4XX", "5XX"]
+    );
+  });
+
+  it("emit diagnostic if status code range is not supported", async () => {
+    const diagnostics = await diagnoseOpenApiFor(
+      `
+      op read(): {@minValue(455) @maxValue(495) @statusCode _: int32, content: string};
```

Comment:
Did we consider simply make this syntax shorthand for `455 | 456 | ... | 495`?

Reply:
you mean in the resulting openapi have all the individual status code?

Reply:
The same as if you manually wrote `201 | 202` etc. Maybe that would make sense as the behavior if you turned the error into a warning.

Reply:
yeah I did think of doing that but the openapi3 might become way to verbose so I think that might need to be a follow up feature request if we ever get one